packages/nuxt/src/pages/runtime/page.ts

```diff
@@ -191,11 +191,14 @@ export default defineComponent({
                 }
 
                 const routerComponentType = routeProps.Component.type as any
-                let PageRouteProvider = routerProviderLookup.get(routerComponentType)
+                const routeProviderKey = import.meta.dev ? routerComponentType.name || routerComponentType.__name : routerComponentType
+                let PageRouteProvider = import.meta.client ? _routeProviders.get(routeProviderKey) : undefined
 
                 if (!PageRouteProvider) {
                   PageRouteProvider = defineRouteProvider(routerComponentType.name || routerComponentType.__name)
-                  routerProviderLookup.set(routerComponentType, PageRouteProvider)
+                  if (import.meta.client) {
+                    _routeProviders.set(routeProviderKey, PageRouteProvider)
+                  }
```

Comment:
There will only run on the client side, would it be okay if we omitted this check?